Development data and sql queries that will create the table and populate with demo data

debasishbsws opened the pull request.

debasishbsws commented:
@Flanker-shyam sql queries check karna

debasishbsws commented on dev-database.sql:
```diff
@@ -48,4 +50,37 @@ ALTER TABLE `institute_course` ADD FOREIGN KEY (`course_id`) REFERENCES `course`
+
+
+-- Populate `question_paper` table
+INSERT INTO `question_paper` (`url`, `year`, `semester`, `exam_type`, `title`, `institute_id`, `course_id`)
+VALUES ('Question Paper URL 1', '2022', 'Spring', 'ENDSEM', 'DEMO Question Paper Title 1', 1, 1),
+       ('Question Paper URL 2', '2022', 'Spring', 'MIDSEM', 'DEMO Question Paper Title 1', 1, 1),
+       ('Question Paper URL 2', '2022', 'Spring', 'MIDSEM', 'DEMO Question Paper Title 2', 1, 1),
+       ('Question Paper URL 3', '2022', 'Autumn', 'ENDSEM', 'DEMO Question Paper Title 3', 1, 1),
+       ('Question Paper URL 1', '2021', 'Autumn', 'ENDSEM', 'DEMO Question Paper Title 1', 1, 1),
+       ('Question Paper URL 4', '2021', 'Spring', 'ENDSEM', 'DEMO Question Paper Title 4', 1, 2);
```
proper link should be added after merge

Flanker-shyam reviewed (commented):
taking too many details is redundant, why we need street address??

and also how to gonna handle google map thing?? it's not clear, I guess it is redundant too.

debasishbsws commented:
> taking too many details is redundant, why we need street address??
> 
> and also how to gonna handle google map thing?? it's not clear, I guess it is redundant too.

Yup we may not need the street and pin etc feel free to add suggestions in the comment.

And for the Google map thing, we can use the Google Maps Geocoding API. 
an example code that uses place id to get an address. (AI generated) may need some changes

```
const axios = require('axios');

async function getAddressDetails(googleMapsLink) {
  try {
    // Extract the place ID from the Google Maps link
    const [, query] = googleMapsLink.split('/maps/');
    const placeId = query.split('?')[0];

    // Make a request to the Geocoding API
    const response = await axios.get('https://maps.googleapis.com/maps/api/geocode/json', {
      params: {
        place_id: placeId,
        key: 'YOUR_API_KEY' // Replace with your Google Maps API key
      }
    });

    // Extract the formatted address from the API response
    const { results } = response.data;
    if (results.length > 0) {
      const formattedAddress = results[0].formatted_address;
      return formattedAddress;
    } else {
      throw new Error('No address found');
    }
  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Example usage
const googleMapsLink = 'https://goo.gl/maps/TxPD2nAYrVkFSAfc9';
getAddressDetails(googleMapsLink)
  .then(address => {
    console.log('Address:', address);
  })
  .catch(error => {
    console.error('Error:', error.message);
  });
```

Flanker-shyam commented:
seems fine, Yepp we can integrate that, but that the similar ques, what is the purpose of taking map location, In context to our application?? 
taking state and country should be sufficient for us to differentiate between institutes.

Flanker-shyam commented:
That too, user can take enter comfortably.

debasishbsws commented:
Okay, I will not be needed for our application IG. So we do not need the Address table and just take the country, state info and save it to the institute table it self.

Flanker-shyam commented:
Yes that would be fine !

debasishbsws commented:
Added the docker compose file that will create the MySQL server as a docker container and run the `dev-database.sql` to create the demo schema and fill it with demo data. also run a `phpmyadmin` server on `localhost:8080` we can use to connect to the database.

Flanker-shyam reviewed (approved):
All good !!
